packages/web-components/src/components/va-radio/va-radio.tsx

```diff
@@ -220,48 +236,77 @@ export class VaRadio {
-      headerAriaDescribedby 
+      headerAriaDescribedby,
+      useFormsPattern,
+      formHeadingLevel,
+      formHeading,
     } = this;
     const ariaLabel = label + (required ? ' required' : '');
     const HeaderLevel = this.getHeaderLevel();
     const headerAriaDescribedbyId = headerAriaDescribedby ? 'header-message' : null;
+    const ariaLabeledByIds = 
+    `${useFormsPattern && formHeading ? 'form-question' : ''} ${ 
+      useFormsPattern ? 'form-description' : ''} ${
+      useFormsPattern && label ? 'input-label' : ''}`.trim() || null;
 
     if (uswds) {
       const legendClass = classnames({
         'usa-legend': true,
         'usa-label--error': error
       });
+
+      const isFormsPattern = useFormsPattern === 'single' || useFormsPattern === 'multiple' ? true : false;
+
+      let formsHeading = null;
+      if (isFormsPattern) {
+        const HeaderLevel = getHeaderLevel(formHeadingLevel);
+        formsHeading = (
+          <Fragment>
+            {formHeading &&
+              <HeaderLevel id="form-question" part="form-header">
+                {formHeading}
+              </HeaderLevel>
+            }
+            <div id="form-description">
+              <slot name="form-description"></slot>
+            </div>
+          </Fragment>
+        )
+      }
       return (
-        <Host aria-invalid={error ? 'true' : 'false'} aria-label={ariaLabel}>
-          <fieldset class="usa-fieldset">
-            <legend class={legendClass} part="legend">
-              {HeaderLevel ? (
-                <HeaderLevel part="header" aria-describedby={headerAriaDescribedbyId}>{label}</HeaderLevel>
-              ) : (
-                label
-              )}
-              {headerAriaDescribedby && (
-                <span id="header-message" class="sr-only">
-                  {headerAriaDescribedby}
-                </span>
-              )}
-              {required && (
-                <span class="usa-label--required" part="required">
-                  {' '}
-                  {i18next.t('required')}
-                </span>
-              )}
-            </legend>
-            {hint && <span class="usa-hint">{hint}</span>}
-            <span class="usa-error-message" role="alert">
-              {error && (
-                <Fragment>
-                  <span class="usa-sr-only">{i18next.t('error')}</span> {error}
-                </Fragment>
-              )}
-            </span>
-            <slot></slot>
-          </fieldset>
+        <Host aria-invalid={error ? 'true' : 'false'} aria-label={ariaLabel} aria-labelledby={ariaLabeledByIds}>
+          {formsHeading}
+          <div class="input-wrap">
+            <fieldset class="usa-fieldset">
+              <legend class={legendClass} part="legend">
+                {HeaderLevel ? (
+                  <HeaderLevel part="header" aria-describedby={headerAriaDescribedbyId}>{label}</HeaderLevel>
+                ) : (
+                  label
+                )}
```

Comment:
Just want to check my understanding here. If a user sets the props `labelHeaderLevel`, `useFormsPattern`, and `formHeadingLevel`, then this component would potentially render two separate headers: One right before `div.input-wrap` and one inside the `legend`. Is this something we want, or, is it something we need to handle?

Reply:
Good point! @danbrady, could you please confirm whether this was discussed during the design process?

Reply:
I haven't been involved in this effort, so I don't know how much help I can be here. Looking back at the [initial ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2269) it says:

> useFormsPattern?: string. Accepts either "single" or "multiple" to indicate if the form is a single input or will have multiple inputs. Enabling this will add a heading and description for integrating into the forms pattern.
**Enabling this will remove headings/descriptions from fieldset, legend, these will be placed outside.**

I'm not sure if that answer your question though. @rsmithadhoc or @jamigibbs can you shed some light here?

Reply:
The ticket description answers the question. Thanks Dan